### PR TITLE
CI: Fix cargo cache

### PR DESCRIPTION
**Pull request by General-Beck**
Blocking waiting for file lock on the registry index

### CODE REVIEW

File: scripts/gitlab-build.sh [hunk 1]
```diff
@@ -48,6 +48,8 @@ set_env_win () {
   echo "@ signtool sign /f "\%"1 /p "\%"2 /tr http://timestamp.comodoca.com /du https://parity.io "\%"3" > sign.cmd
 }
 build () {
+  echo "Remove index"
+  rm -rf cargo/registry/index/*.
```
Comment (debris) [inline scripts/gitlab-build.sh]: I guess you meant, `.cargo/registry/index`? That's a terrible idea! Cargo will have to redownload it's crates index and cargo builds may interfere with each.

Comment (5chdn) [reply scripts/gitlab-build.sh]: Cargo cache is in `cargo/` #7921: https://github.com/paritytech/parity/pull/7921/files

The path "is correct", but I can't evaluate if this is the correct way to address this. Problem is that windows builds fail due to a build lock even though there is no other build running.

https://gitlab.parity.io/parity/parity/-/jobs/80511

Please advise :)

Comment (kirushik) [reply scripts/gitlab-build.sh]: @General-Beck Let's wrap this in the conditional `if [[ "windows" = $IDENT ]]` for now?
I guess the actual root cause is our cache key selection, but I would do that as a part of enhancing the #7988

Comment (debris) [reply scripts/gitlab-build.sh]: @5chdn I believe this is not a correct way of addressing the problem

**Pull request merged**